umapi_client/functional.py

```diff
@@ -308,6 +308,32 @@ def delete_account(self):
         self.append(removeFromDomain={})
         return None
 
+    def split_groups(self, index, verb, group_type, max_groups):
```

Comment:
see comments in the connections file.  You should only have arguments `self` and `max_groups` and do a single iteration where you test for and fix long groups.